### PR TITLE
OCPBUGS-100274: Replace hardcoded namespace with configurable OPERATOR_NAMESPACE

### PR DESCRIPTION
**Pull request by bharti-0211**
Running `OPERATOR_NAMESPACE=clusterresourceoverride-operator make e2e` fails immediately with:
`kubectl -n openshift-cluster-resource-override rollout status -w deployment/clusterresourceoverride-operator                                                                                                   
Error from server (NotFound): namespaces "openshift-cluster-resource-override" not found                                                                                                                       
make: *** [Makefile:136: e2e] Error 1`

With this PR changes, if the user provides `OPERATOR_NAMESPACE` that value is picked up.

## Summary by CodeRabbit

* **New Features**
  * Configure the operator namespace through a command-line option or environment variable.
  * Deployment manifests automatically use the selected namespace.
  * End-to-end and upgrade tests consistently use the configured namespace, with a default fallback.

* **Bug Fixes**
  * Custom namespace values are validated against Kubernetes naming requirements.
  * Empty namespace settings safely fall back to the default.
  * Deployment and upgrade testing now remain consistent when using custom namespaces.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,22 @@ IMAGE_VERSION := 5.0
...
+# (or a single character [a-z0-9]). Reject anything else early so that malformed
+# values do not reach 'sed', 'kubectl', or the Go test binary.
+# Pass the value via an env variable to avoid shell quoting issues on edge-case chars.
+_ns_check := $(shell NS_VAL='$(OPERATOR_NAMESPACE)' sh -c 'printf "%s" "$$NS_VAL" | grep -Eq "^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$$" && echo ok || echo bad')
+ifneq ($(_ns_check),ok)
+$(error OPERATOR_NAMESPACE "$(OPERATOR_NAMESPACE)" is not a valid Kubernetes namespace name. \
+  Must be lowercase alphanumeric and hyphens only, 1-63 characters, \
+  starting and ending with an alphanumeric character.)
+endif
+export OPERATOR_NAMESPACE
```
[the remark this answers is not in the document]
Comment (bharti-0211) [reply Makefile:25]: addressed

**Comment by bharti-0211** [comment]
@joelsmith @maxcao13  could you please review this one? I would appreciate some feedback or suggestions. Thanks.

File: test/e2e/setup_test.go [hunk 2]
```diff
@@ -37,9 +42,37 @@ func TestMain(m *testing.M) {
...
+	// API call or is assigned to shared test state. Use the same IsDNS1123Label
+	// function Kubernetes itself uses to validate namespace names — this ensures
+	// our check stays in sync with Kubernetes validation rules automatically.
+	if errs := validation.IsDNS1123Label(ns); len(errs) > 0 {
+		panic(fmt.Sprintf(
+			"invalid operator namespace %q: %s",
+			ns, strings.Join(errs, "; "),
+		))
+	}
+
```
Comment (maxcao13) [inline test/e2e/setup_test.go]: I don't think this validation is needed, the apiserver should reject invalid names.

File: Makefile [hunk 3]
```diff
@@ -21,7 +21,29 @@ IMAGE_VERSION := 5.0
...
+  else \
+    awk '{n=length($$0); if(n<1||n>63)exit; if($$0!~/^[a-z0-9]([a-z0-9-]*[a-z0-9])?$$/){exit}; print "ok"}' $(_ns_tmpf); \
+  fi; \
+  rm -f $(_ns_tmpf))
+ifneq ($(_ns_check),ok)
+$(error OPERATOR_NAMESPACE "$(OPERATOR_NAMESPACE)" is not a valid Kubernetes namespace name. \
+  Must be lowercase alphanumeric and hyphens only, 1-63 characters, \
+  starting and ending with an alphanumeric character.)
+endif
+
```
Comment (maxcao13) [inline Makefile]: Similar to the other comment, I don't think we need to do all this validation.

A simple:
```
OPERATOR_NAMESPACE ?= openshift-cluster-resource-override
```
suffices, unless there's a good reason that I don't know about?

File: Makefile [hunk 4]
```diff
@@ -116,6 +116,9 @@ deploy:
 	cp manifests/stable/clusterresourceoverride.crd.yaml $(KUBE_MANIFESTS_DIR)/
 	cp $(ARTIFACTS)/registry-env.yaml $(KUBE_MANIFESTS_DIR)/
 
+	# Inject the operator namespace into all copied manifests
+	find $(KUBE_MANIFESTS_DIR) -name "*.yaml" | xargs sed -i "s/openshift-cluster-resource-override/$(OPERATOR_NAMESPACE)/g"
```
[the remark this answers is not in the document]
Comment (maxcao13) [reply Makefile:123]: Is this not overengineering? I find it hard to believe people would put in some invalid characters for a namespace in an e2e test.

**Review by maxcao13 — COMMENTED** [review]
sorry for being tardy on this, I don't agree with coderabbit's comments, IMO it just makes the code harder to read about when this should be a simple fix

File: test/e2e/setup_test.go (hunk 2 above)
Comment (bharti-0211) [reply test/e2e/setup_test.go]: Agree.

File: Makefile (hunk 3 above)
Comment (bharti-0211) [reply Makefile]: No. Simplified it.

**Comment by bharti-0211** [comment]
@maxcao13  Sorry for the delay! Thanks for your feedback. Both changes implemented — please take a look.

**Comment by bharti-0211** [comment]
@joelsmith @maxcao13 Since both the `lgtm` and `verified` labels are required for merging, I’ve added the verified label - not sure it should be added by me. Can you help me here?

**Comment by joelsmith** [comment]
Thanks for adding this!
/lgtm
